Source/CBL_Router+Handlers.m

```diff
@@ -769,15 +779,22 @@ - (CBLStatus) update: (CBLDatabase*)db
     else
         *outRev = [db putRevision: rev prevRevisionID: prevRevID
                     allowConflict: allowConflict
-                           status: &status];
+                           status: &status
+                            error: outError];
+
+    if (CBLStatusIsError(status)) {
+        if (outError && !*outError)
```

Comment:
You can put the assignment to `*outError` right after the `-putLocalRevision:` call, since that's the only one that would set `status` but not `*outError`.